lib/ops/opsservice/configure.go

```diff
@@ -811,82 +801,73 @@ type masterConfig struct {
 	config planetConfig,
 	secretsPackage, configPackage loc.Locator,
 ) error {
-	if server := installOrExpand.InstallExpand.Servers.FindByIP(master.AdvertiseIP); server != nil {
+	if server := config.installExpand.InstallExpand.Servers.FindByIP(config.server.AdvertiseIP); server != nil {
 		config.dockerRuntime = server.Docker
 	}
 
-	err := s.configurePlanetServer(master, installOrExpand, config)
+	err := s.configurePlanetServer(config)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	certAuthorityPackage, err := s.planetCertAuthorityPackage()
-	if err != nil {
-		return trace.Wrap(err)
-	}
-
-	master.PackageSet.AddArchivePackages(*certAuthorityPackage)
-	master.PackageSet.AddArchivePackage(secretsPackage,
-		map[string]string{pack.InstalledLabel: pack.InstalledLabel})
-	master.PackageSet.AddArchivePackage(
-		config.planetPackage, map[string]string{
-			pack.InstalledLabel: pack.InstalledLabel,
-			pack.PurposeLabel:   pack.PurposeRuntime,
-		})
-	master.PackageSet.AddArchivePackage(configPackage,
-		pack.ConfigLabels(config.planetPackage, pack.PurposePlanetConfig))
 	return nil
 }
 
 func (s *site) configurePlanetNode(
-	node *ProvisionedServer,
-	installOrExpand ops.SiteOperation,
 	config planetConfig,
 	secretsPackage, configPackage loc.Locator,
 ) error {
-	if server := installOrExpand.InstallExpand.Servers.FindByIP(node.AdvertiseIP); server != nil {
+	if server := config.installExpand.InstallExpand.Servers.FindByIP(config.server.AdvertiseIP); server != nil {
 		config.dockerRuntime = server.Docker
 	}
 
-	err := s.configurePlanetServer(node, installOrExpand, config)
+	err := s.configurePlanetServer(config)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	node.PackageSet.AddArchivePackage(secretsPackage, map[string]string{
-		pack.InstalledLabel: pack.InstalledLabel,
-	})
-	node.PackageSet.AddArchivePackage(config.planetPackage, map[string]string{
-		pack.InstalledLabel: pack.InstalledLabel,
-		pack.PurposeLabel:   pack.PurposeRuntime,
-	})
-	node.PackageSet.AddArchivePackage(configPackage,
-		pack.ConfigLabels(config.planetPackage, pack.PurposePlanetConfig))
 	return nil
 }
 
-func (s *site) getPlanetConfigPackage(
-	node *ProvisionedServer,
-	installOrExpand ops.SiteOperation,
-	config planetConfig,
-	manifest schema.Manifest) (*ops.RotatePackageResponse, error) {
-	profile, err := manifest.NodeProfiles.ByName(node.Role)
+func (s *site) getPlanetConfigPackage(config planetConfig) (*ops.RotatePackageResponse, error) {
+	args, err := s.getPlanetConfig(config)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	reader, err := pack.GetConfigPackage(s.packages(), config.planetPackage, config.configPackage, args)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	labels := map[string]string{
+		pack.PurposeLabel:     pack.PurposePlanetConfig,
+		pack.ConfigLabel:      config.planetPackage.ZeroVersion().String(),
+		pack.AdvertiseIPLabel: config.server.AdvertiseIP,
+		pack.OperationIDLabel: config.installExpand.ID,
```

Comment:
I'm not sure I understand why AdvertiseIP and OperationID are being added as labels / what purpose they serve? Usually when I think of a labelling system, I think of it as a way to attach selectors to retrieve a set or individual object. To me, I'm not sure what embedding server config in the label is meant to do, and the operation to me seems more likely it should maybe be metadata, which we may not have support for?

Reply:
This is a good question. I think `OperationID` was added as a means of being able to pull the package during init/bootstrap step of the corresponding operation (@r0mant correct me if I'm wrong), I would think advertise IP serves similar purpose but I'm not completely sure. I will see if I can do away with that (note though that this PR is a backport of existing changes - hence I'd avoid making too many deviations).

Reply:
As a backport I wouldn't worry about it then.

Reply:
These labels are used by install FSM when pulling packages for specific operation/server, yes.